### DISCUSSION TITLE
Trying to decrypt a value encrypted by Postgres pgp_sym_encrypt

### DISCUSSION THREAD

**Discussion by eladchen** [openpgpjs/openpgpjs#1698] [Q&A]
Hi,

Take a look at the following snippet, it defines a buffer representing the db value (`encryptedValueFromDatabase`)

I'm using Postgres 13, and according to the [docs](https://www.postgresql.org/docs/13/pgcrypto.html) (F.25.3.8.1. cipher-algo)
when using `pgp_sym_encrypt` the default cipher is aes128 with no compression.

```js
const openpgp = require('openpgp');

// Key used for encryption
const KEY = 'REDACTED'; // can't share this

async function decryptByteArray() {
  // Encrypted value obtained from PostgreSQL's pgp_sym_encrypt
  const encryptedValueFromDatabase = Buffer.from('C30D04070302CB64B62383A829D070D24201863B5854428DCB9FFB62FFD289DE9CDFDE09B42F2C0FEAE13432EE8C7868C7E5432F63D793A36AD801E87946F95C767B1A67529818646032BC453A3F8C38BF90D4', 'hex');
  const message = await openpgp.readMessage({ binaryMessage: encryptedValueFromDatabase });
  const result = await openpgp.decryptSessionKeys({
    message,
    passwords: [KEY],
    config: {
      preferredSymmetricAlgorithm: openpgp.enums.symmetric.aes128,
      // preferredCompressionAlgorithm: openpgp.enums.compression.bzip2,
    },
  });
  const decrypted = Buffer.from(result[0].data).toString('utf-8');

  console.log('decrypted: %s', decrypted);
}

return decryptByteArray();
```

I'm using openpgp@5.11.0.

P.S

- I verified that using the same key with Postgres decryption function works (`pgp_sym_decrypt`), but that is on the DB level and doesn't help.

- I have also tried to encrypt the value using openpgp, it seems the created byte array it creates is far smaller then the one created by `pgp_sym_encrypt`

**Comment by larabr** [comment]
Hi, what is the question? Do you get an error, or?

**Comment by eladchen** [comment]
The decrypt fails.. any idea what am I doing wrong?

**Reply by larabr** [reply]
What's the error message?
The code as posted cannot run, is `KEY` an array? Make sure you are passing either a string, or array of strings as `passwords`.

**Reply by twiss** [reply]
Hey :wave: In addition to what Lara said, if you want to decrypt the message (rather than just the session keys), you need to use `openpgp.decrypt` rather than `openpgp.decryptSessionKeys`. (Also, the `config` isn't necessary, the algorithm used is specified in the message.)

Edit: looking at the docs, you might also have to enable the `sess-key` option (set it to `1`), otherwise OpenPGP.js will most likely complain by default due to a legacy symmetric encryption method.

**Reply by eladchen** [reply]
@larabr its just a typo, should have been a string. I edited the mistake.

**Reply by eladchen** [reply]
@twiss Any example? Would love it if you can point me to a test case using the suggested solution, plus the  `sess-key`.

**Reply by twiss** [reply]
See the second half of https://github.com/openpgpjs/openpgpjs#encrypt-and-decrypt-uint8array-data-with-a-password.
You can remove the `format: 'binary'` from the example if you want to get the decrypted message as a Unicode string.

**Reply by eladchen** [reply]
It seems like you meant I should use `sess-key` when Postgres does the encryption?

There are well over 500 millions records that already exists, anything I can do using openpgp?

**Reply by larabr** [reply]
The encrypted message you shared looks fine, it should decrypt. What error message are you getting?

**Reply by rene-leanix** [reply]
@eladchen, did you solve this in the end?

**Reply by eladchen** [reply]
@rene-leanix Yes
